src/D2L.Security.OAuth2/Keys/UnsignedToken.cs

```diff
@@ -1,75 +1,97 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.Collections.ObjectModel;
+using System.Linq;
 using System.Security.Claims;
 
 namespace D2L.Security.OAuth2.Keys {
 
-	/// <summary>
-	/// A token that has yet to be signed
-	/// </summary>
-	public sealed class UnsignedToken {
-		private readonly string m_issuer;
-		private readonly string m_audience;
-		private readonly ReadOnlyCollection<Claim> m_claims;
-		private readonly DateTime m_notBefore;
-		private readonly DateTime m_expiresAt;
+    /// <summary>
+    /// A token that has yet to be signed
+    /// </summary>
+    public sealed class UnsignedToken {
+        private readonly string m_issuer;
+        private readonly string m_audience;
+        private readonly IReadOnlyDictionary<string, object> m_claims;
+        private readonly DateTime m_notBefore;
+        private readonly DateTime m_expiresAt;
 
-		/// <summary>
-		/// Constructs a new <see cref="UnsignedToken"/> instance
-		/// </summary>
-		/// <param name="issuer">Identifies the issuer of the JWT; this is the 'iss' claim</param>
-		/// <param name="audience">Identifies the recipients that the JWT is itended for; this is the 'aud' claim</param>
-		/// <param name="claims">The token's claims</param>
-		/// <param name="notBefore">When the token becomes valid; this is the 'nbf' claim</param>
-		/// <param name="expiresAt">When the token expires; this is the 'exp' claim</param>
-		public UnsignedToken(
-			string issuer,
-			string audience,
-			IList<Claim> claims,
-			DateTime notBefore,
-			DateTime expiresAt
-		) {
-			m_issuer = issuer;
-			m_audience = audience;
-			m_claims = new ReadOnlyCollection<Claim>( claims );
-			m_notBefore = notBefore;
-			m_expiresAt = expiresAt;
-		}
+        /// <summary>
+        /// Constructs a new <see cref="UnsignedToken"/> instance
+        /// </summary>
+        /// <param name="issuer">Identifies the issuer of the JWT; this is the 'iss' claim</param>
+        /// <param name="audience">Identifies the recipients that the JWT is itended for; this is the 'aud' claim</param>
+        /// <param name="claims">The token's claims</param>
+        /// <param name="notBefore">When the token becomes valid; this is the 'nbf' claim</param>
+        /// <param name="expiresAt">When the token expires; this is the 'exp' claim</param>
+        public UnsignedToken(
+            string issuer,
+            string audience,
+            IList<Claim> claims,
+            DateTime notBefore,
+            DateTime expiresAt
+        ) {
+            m_issuer = issuer;
+            m_audience = audience;
+            m_claims = claims.ToDictionary( t => t.Type, t => (object)t.Value );

```

Comment:
I think I imagined this constructor going away. This is maybe fine. Just noting this could throw a "key already added" exception - avoiding having to do this conversion ourselves removes the chance of that "oops" case from our API surface. No particularly strong feelings though.

Reply:
Will drop the old constructor.